config/urfavecli/clicontext/user.go

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (C) 2019 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+// Package urfavecli/clicontext is an adapter to load configuration from urfave/cli.v1 Context
+
+package clicontext
+
+import (
+	"os"
+	"path"
+
+	"github.com/mysteriumnetwork/node/config"
+	"github.com/mysteriumnetwork/node/logconfig"
+	"github.com/pkg/errors"
+	"gopkg.in/urfave/cli.v1"
+)
+
+var log = logconfig.NewLogger()
+
+// LoadUserConfig determines config location from the context
+// and makes sure that the config file actually exists, creating it if necessary
+func LoadUserConfig(ctx *cli.Context) error {
+	configDir, configFilePath := resolveLocation(ctx)
+	err := createDirIfNotExists(configDir)
+	if err != nil {
+		return err
+	}
+	err = createFileIfNotExists(configFilePath)
+	if err != nil {
+		return err
+	}
+
+	return config.Current.LoadUserConfig(configFilePath)
+}
+
+// LoadUserConfigQuietly like LoadUserConfig, but instead of returning an error,
+// it logs it on a `warn` level.
+// `error` is specified as a return to adhere to `cli.BeforeFunc` for convenience.
+func LoadUserConfigQuietly(ctx *cli.Context) error {
```

Comment:
don't need the return value here

Reply:
Need it to implement `BeforeFunc` type from `urfave/cli`:

```go
type BeforeFunc func(*Context) error
```

This is to simplified usage:

```go
return &cli.Command{
		Name:   cliCommandName,
		Usage:  "Starts a CLI client with a Tequilapi",
		Before: clicontext.LoadUserConfigQuietly,
		...
}
```